code/__HELPERS/path.dm

```diff
@@ -135,34 +137,46 @@
 	src.mintargetdist = mintargetdist
 	src.simulated_only = simulated_only
 	src.avoid = avoid
+	src.skip_first = skip_first
 	src.diagonal_safety = diagonal_safety
+	src.on_finish = on_finish
+
+/datum/pathfind/Destroy(force, ...)
+	. = ..()
+	SSpathfinder.active_pathing -= src
+	SSpathfinder.currentrun -= src
+	if(on_finish)
+		on_finish.Invoke(null)
```

Comment:
Would this not cause bots to alert failure to find their path every time it's qdel'ed, like in finished()?

Reply:
if the callback hasn't been invoked by the time qdel is called, yes